src/edge_kernels/VOFAdvectionEdgeAlg.C

```diff
@@ -39,27 +42,51 @@ VOFAdvectionEdgeAlg::VOFAdvectionEdgeAlg(
   massFlowRate_ = get_field_ordinal(
     meta, (useAverages) ? "average_mass_flow_rate" : "mass_flow_rate",
     stk::topology::EDGE_RANK);
-  density_ =
-    get_field_ordinal(realm.meta_data(), "density", stk::mesh::StateNP1);
+  massForcedFlowRate_ =
+    get_field_ordinal(meta, "mass_forced_flow_rate", stk::topology::EDGE_RANK);
+  density_ = get_field_ordinal(meta, "density", stk::mesh::StateNP1);
+  velocity_n_ = get_field_ordinal(meta, "velocity", stk::mesh::StateN);
+
+  std::map<PropertyIdentifier, MaterialPropertyData*>::iterator itf =
+    realm_.materialPropertys_.propertyDataMap_.find(DENSITY_ID);
+
+  if (itf == realm_.materialPropertys_.propertyDataMap_.end()) {
+    density_liquid_ = 1000.0;
```

Comment:
What is the use case for this?  Don't we want a missing property map entry to throw as well?

Reply:
This is for the unit testing of VOFAdvectionEdgeAlg which would just throw unless all the piping for property maps were introduced. I could also make it something that is passed into the routine directly instead.